src/opts.c

```diff
@@ -298,8 +298,13 @@ _hostnameMatchesSubjectAltName(char *hostname, X509 *cert)
         if (namePart->type == GEN_DNS)
         {
             foundAnyEntry = true;
-            foundMatch = _hostnameMatches((char*) ASN1_STRING_data(namePart->d.uniformResourceIdentifier),
-                                           hostname);
+
+            if (!(natsOpenSSL_Version(1, 1, 0, OPENSSL_VERSION_NUMBER)))
```

Comment:
Unfortunately, that does not work since you are detecting at runtime but then have in the code reference to a function that may not even be known at the time of the building of the library. (see travis build report that fails with openssl pre 1.1.0).

Reply:
Yes exactly! The check should be in compile time. I will fix it if later when I have time if you think it is worth to be implemented. Thanks.